### ISSUE TITLE
Launching Ethermint / Tendermint / ABCI

### ISSUE THREAD

**Issue by berrehili** [tendermint/tendermint#1484]
**BUG REPORT **

Hello again everyone! 

Since I've been trying to compile Ethermint from source without massive success, I then followed through the binary files. I think most of the steps I needed to go through were correctly followed. 

**Ethermin & Ethereum installed:**
`ethermint version
ethermint:  0.5.3-b72a1eef
go-ethereum:  1.6.7-stable`

**Tendermint version** 
`Tendermint version 0.18.0-4930b61a`

**ABCI version** 
`abci-cli version 0.7.1`

**Environment:** 
Ubuntu 16.04
AMD

**What happened**:
First I initiated the datadir and home for Ethermint and Tendermint:
`tendermint init --home ~/.ethermint/tendermint
ethermint --datadir ~/.ethermint init`

Then I launch Tendermint which takes me to the first errors:

`root@serveur-9:~/.ethermint/tendermint/config# tendermint --home ~/.ethermint/tendermint node`

> I[04-21|22:07:17.963] Starting multiAppConn                        module=proxy impl=multiAppConn
> I[04-21|22:07:17.963] Starting socketClient                        module=abci-client connection=query impl=socketClient
> E[04-21|22:07:17.964] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:20.965] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:23.965] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:26.966] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:29.967] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:32.967] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query
> E[04-21|22:07:35.968] abci.socketClient failed to connect to tcp://127.0.0.1:46658.  Retrying... module=abci-client connection=query

The errors changes when I launch Ethermint which gives this:

> root@serveur-9:/home/ubuntu# ethermint --datadir ~/.ethermint --rpc --rpcaddr=0.0.0.0 --ws --wsaddr=0.0.0.0 --rpcapi eth,net,web3,personal,admin
> INFO [04-21|22:07:34] Starting peer-to-peer node               instance=ethermint/v1.6.7-stable/linux-amd64/go1.8.3
> INFO [04-21|22:07:34] Allocated cache and file handles         database=/root/.ethermint/ethermint/chaindata cache=128 handles=1024
> INFO [04-21|22:07:34] Initialised chain configuration          config="{ChainID: 15 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: <nil> EIP155: 0 EIP158: 0 Metropolis: <nil> Engine: unknown}"
> WARN [04-21|22:07:34] Ethash used in fake mode 
> INFO [04-21|22:07:34] Initialising Ethereum protocol           versions="[63 62]" network=1
> INFO [04-21|22:07:34] Loaded most recent local header          number=0 hash=2b72f9…d468ac td=64
> INFO [04-21|22:07:34] Loaded most recent local full block      number=0 hash=2b72f9…d468ac td=64
> INFO [04-21|22:07:34] Loaded most recent local fast block      number=0 hash=2b72f9…d468ac td=64
> INFO [04-21|22:07:34] Starting P2P networking 
> INFO [04-21|22:07:34] RLPx listener up                         self="enode://195a1461cdcb97362399213cabca967f80fee28a14cfdb3867b0188d3e5f68140f744b99e2c5c60f4bd02072cdffa8ac305669b057762bd458c2a9d27377125f@[::]:30303?discport=0"
> INFO [04-21|22:07:34] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"
> INFO [04-21|22:07:34] HTTP endpoint opened: http://0.0.0.0:8545 
> INFO [04-21|22:07:34] WebSocket endpoint opened: ws://0.0.0.0:8546 
> INFO [04-21|22:07:34] IPC endpoint opened: /root/.ethermint/geth.ipc 
> INFO [04-21|22:07:36] Starting ABCIServer                      module=abci-server impl=ABCIServer
> INFO [04-21|22:07:36] Waiting for new connection...            module=abci-server
> INFO [04-21|22:07:37] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"
> INFO [04-21|22:07:38] Accepted a new connection                module=abci-server
> INFO [04-21|22:07:38] Waiting for new connection...            module=abci-server
> INFO [04-21|22:07:38] Accepted a new connection                module=abci-server
> INFO [04-21|22:07:38] Waiting for new connection...            module=abci-server
> INFO [04-21|22:07:38] Accepted a new connection                module=abci-server
> INFO [04-21|22:07:38] Waiting for new connection...            module=abci-server
> ERROR[04-21|22:07:38] Connection error                         module=abci-server error="Error reading message: Varint overflow"
> INFO [04-21|22:07:40] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"
> INFO [04-21|22:07:43] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"
> INFO [04-21|22:07:46] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"
> INFO [04-21|22:07:49] Waiting for tendermint endpoint to start err="Post http://localhost:46657/status: dial tcp 127.0.0.1:46657: getsockopt: connection refused"

And then Tendermint gives that:

> I[04-21|22:07:38.969] Starting socketClient                        module=abci-client connection=mempool impl=socketClient
> I[04-21|22:07:38.969] Starting socketClient                        module=abci-client connection=consensus impl=socketClient
> E[04-21|22:07:38.970] Stopping abci.socketClient for error: EOF    module=abci-client connection=query
> I[04-21|22:07:38.971] Stopping socketClient                        module=abci-client connection=query impl=socketClient

All the ports are open, I suspect this error to be the real issue: 

> ERROR[04-21|22:07:38] Connection error module=abci-server error="Error reading message: Varint overflow"

**What you expected to happen**:  Having the block creation happening.

Thank you for your support! :)

**Comment by melekes** [comment]
"The latest version of Ethermint only works with Tendermint up to 0.14.0." https://github.com/tendermint/ethermint/blob/master/README.md

Sorry about that.

**Comment by berrehili** [comment]
HI @melekes ! Thank you for your feedback. 

So I just downgraded to tendermint 0.14.0-'88f5f21d'.

First, I generate a new genesis state : 
> ethermint --datadir ~/.ethermint --with-tendermint init

Which worked fine, but still couldn't launch a working node. So I discovered that I had to delete all the files in  ~/.ethermint and re-launch a new genesis state. Which worked perfectly fine. 

Are you planning to work on a new version of Ethermint that is compatible with latest version of Tendermint?
